eks_fargate/datadog_checks/eks_fargate/eks_fargate.py

```diff
@@ -0,0 +1,16 @@
+# (C) Datadog, Inc. 2020
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datadog_checks.base import AgentCheck
+import os
+
+class EksFargateCheck(AgentCheck):
+    def check(self, instance):
+        virtual_node = os.getenv("DD_KUBERNETES_KUBELET_NODENAME")
+        pod_name = os.getenv("HOSTNAME")
+        # Only submit the heartbeat metric for fargate virtual nodes.
+        if "fargate" in virtual_node:
```

Comment:
I think we have no guarantee that `virtual_node` will always contain `fargate` in future eks-fargate releases right?
I'd rather verify `DD_KUBERNETES_KUBELET_NODENAME` and `DD_EKS_FARGATE` if one of them is not set we don't run the check (with corresponding error and log messages to explain why), and we run the check regardless of the `virtual_node` value if `DD_KUBERNETES_KUBELET_NODENAME` exists and is_affirmative(`DD_EKS_FARGATE`). wdyt?

Reply:
Unless users manually move the check, it will only be run if DD_EKS_FARGATE is set to true.
Per: https://github.com/DataDog/datadog-agent/pull/4699/files#diff-56cf5743b91b6eee2e1c11904e8d065cR19-R20

Because the node name can't be configured for fargate nodes, relying on `fargate-` in the node name is a better guarantee that the workload is indeed running in fargate mode.
If we decide to solely rely on `DD_EKS_FARGATE`, for agents running on regular EKS nodes that have the env var, we would start accounting for them as EKS over Fargate nodes.